coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,651 @@
+
+                    The groups ``line``, ``column``, ``end_line`` and
+                    ``end_column`` don't have to match numbers only, they can
+                    also match nothing, the generated ``Result`` is filled
+                    automatically with ``None`` then for the appropriate
+                    properties.
+                :param severity_map:
+                    A dict used to map a severity string (captured from the
+                    ``output_regex`` with the named group ``severity``) to an
+                    actual ``coalib.results.RESULT_SEVERITY`` for a result.
+                :return:
+                    An iterator returning results.
+                """
+                if isinstance(output, str):
+                    output = (output,)
+
+                for string in output:
+                    for match in re.finditer(output_regex, string):
+                        yield self._convert_output_regex_match_to_result(
+                            match, filename, severity_map=severity_map)
+
+            if options["output_format"] is None:
+                # Check if user supplied a `process_output` override.
+                if not callable(getattr(klass, "process_output", None)):
+                    raise ValueError("`process_output` not provided by given "
+                                     "class {}.".format(repr(klass.__name__)))
+                # No need to assign to `process_output` here, the class mixing
+                # below automatically does that.
+            else:
+                # Prevent people from accidentally defining `process_output`
+                # manually, as this would implicitly override the internally
+                # set-up `process_output`.
+                if hasattr(klass, "process_output"):
+                    raise ValueError("Found `process_output` already defined "
+                                     "by class {}, but {} output-format is "
+                                     "specified.".format(
+                                         repr(klass.__name__),
+                                         repr(options["output_format"])))
+
+                if options["output_format"] == "corrected":
+                    process_output_args = {
+                        key: options[key]
+                        for key in ("diff_message", "diff_severity")
+                        if key in options}
+
+                    process_output = partialmethod(
+                        process_output_corrected, **process_output_args)
+
+                else:
+                    assert options["output_format"] == "regex"
+
+                    process_output_args = {
+                        key: options[key]
+                        for key in ("output_regex", "severity_map")
+                        if key in options}
+
+                    process_output = partialmethod(
+                        process_output_regex, **process_output_args)
```

Comment:
you think that makes things easier? :O
setting the `process_output` is quite embedded into the class, needs options, class functions and the class to mixin later^^